### PR TITLE
fix storage bootstrap for sovereign

### PR DESCRIPTION
**Pull request by raduchis**
## Reasoning behind the pull request
- bootstrap from storage did not work for sovereign
  
## Proposed changes
- fix bootstrap from storage

## Testing procedure
- run a sovereign testnet, stop it and start with db 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by andreibancioiu — APPROVED** [review]

File: epochStart/bootstrap/storageProcess.go [hunk 1]
```diff
@@ -72,7 +72,11 @@ func NewStorageEpochStartBootstrap(args ArgsStorageEpochStartBootstrap) (*storag
 
 	esbConverted, ok := esb.(*epochStartBootstrap)
 	if !ok {
-		return nil, errors.ErrInvalidTypeConversion
+		sovereignESB, ok := esb.(*sovereignChainEpochStartBootstrap)
```
Comment (mariusmihaic) [inline epochStart/bootstrap/storageProcess.go:75]: I think we should go for the clean version and use factories from runTypeComponents actually.

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

File: epochStart/bootstrap/storageProcess.go (hunk 1 above)
Comment (sasurobert) [reply epochStart/bootstrap/storageProcess.go:75]: +1

File: epochStart/bootstrap/storageProcess_test.go [hunk 2]
```diff
@@ -98,17 +101,9 @@ func TestNewStorageEpochStartBootstrap_ShouldWork(t *testing.T) {
 
 		coreComp, cryptoComp := createComponentsForEpochStart()
 		args := createMockStorageEpochStartBootstrapArgs(coreComp, cryptoComp)
-		sesb, err := NewStorageEpochStartBootstrap(args)
-		assert.False(t, check.IfNil(sesb))
-		assert.Nil(t, err)
-	})
-
-	t.Run("should work for sovereign", func(t *testing.T) {
```
Comment (mariusmihaic) [inline epochStart/bootstrap/storageProcess_test.go]: This test for sovereign was deleted, but I think we can still reuse the sovereign chain creation test in a different file

File: epochStart/bootstrap/epochStartBootstrapperFactory.go [hunk 3]
```diff
@@ -14,6 +14,16 @@ func (bcf *epochStartBootstrapperFactory) CreateEpochStartBootstrapper(epochStar
 	return NewEpochStartBootstrap(epochStartBootstrapArgs)
 }
 
+// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for chain run normal
```
Comment (mariusmihaic) [inline epochStart/bootstrap/epochStartBootstrapperFactory.go:17]: Perhaps ?
```suggestion
// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for normal chain operations.
```

Also, regarding this file and `epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go`, can we please add unit tests (at least for some code coverage)?

File: epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go [hunk 4]
```diff
@@ -22,6 +22,26 @@ func NewSovereignEpochStartBootstrapperFactory(esbf EpochStartBootstrapperCreato
 
 // CreateEpochStartBootstrapper creates a new epoch start bootstrapper for chain run sovereign
```
Comment (mariusmihaic) [inline epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:23]:
```suggestion
// CreateEpochStartBootstrapper creates a new epoch start bootstrapper for sovereign chain operations
```

File: epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go [hunk 5]
```diff
@@ -22,6 +22,26 @@ func NewSovereignEpochStartBootstrapperFactory(esbf EpochStartBootstrapperCreato
...
 func (bcf *sovereignEpochStartBootstrapperFactory) CreateEpochStartBootstrapper(epochStartBootstrapArgs ArgsEpochStartBootstrap) (EpochStartBootstrapper, error) {
+	sesb, err := bcf.createSovereignEpochStartBootStrapper(epochStartBootstrapArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	return sesb, nil
+}
+
+// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for chain run normal
```
Comment (mariusmihaic) [inline epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:33]:
```suggestion
// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for sovereign chain operations
```

File: epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go [hunk 6]
```diff
@@ -22,6 +22,26 @@ func NewSovereignEpochStartBootstrapperFactory(esbf EpochStartBootstrapperCreato
 
 // CreateEpochStartBootstrapper creates a new epoch start bootstrapper for chain run sovereign
 func (bcf *sovereignEpochStartBootstrapperFactory) CreateEpochStartBootstrapper(epochStartBootstrapArgs ArgsEpochStartBootstrap) (EpochStartBootstrapper, error) {
+	sesb, err := bcf.createSovereignEpochStartBootStrapper(epochStartBootstrapArgs)
```
Comment (mariusmihaic) [inline epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:25]: I think we can return directly here

Comment (raduchis) [reply epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:25]: fixed

File: epochStart/bootstrap/storageProcess.go (hunk 1 above)
Comment (raduchis) [reply epochStart/bootstrap/storageProcess.go]: refactored

File: epochStart/bootstrap/epochStartBootstrapperFactory.go (hunk 3 above)
Comment (raduchis) [reply epochStart/bootstrap/epochStartBootstrapperFactory.go]: added tests

**Review by mariusmihaic — DISMISSED** [review]

**Review by mariusmihaic — APPROVED** [review]

File: epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go [hunk 7]
```diff
@@ -20,19 +20,29 @@ func NewSovereignEpochStartBootstrapperFactory(esbf EpochStartBootstrapperCreato
...
+
+// CreateStorageEpochStartBootstrapper creates a new storage epoch start bootstrapper for sovereign chain operations
+func (bcf *sovereignEpochStartBootstrapperFactory) CreateStorageEpochStartBootstrapper(epochStartBootstrapArgs ArgsStorageEpochStartBootstrap) (EpochStartBootstrapper, error) {
+	esb, err := bcf.createSovereignEpochStartBootStrapper(epochStartBootstrapArgs.ArgsEpochStartBootstrap)
 	if err != nil {
 		return nil, err
 	}
 
-	sesb, err := NewSovereignChainEpochStartBootstrap(epochStartBootstrapper)
+	epochStartBootstrapArgs.EpochStartBootStrap = esb.epochStartBootstrap
```
Comment (andreibancioiu) [inline epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:35]: This mutates the input `args` / side effect (the alternative would have been to clone it, then mutate it).

Comment (andreibancioiu) [reply epochStart/bootstrap/sovereignEpochStartBootstrapperFactory.go:35]: Oh, now I see, `epochStartBootstrapArgs` is passed by value, all good :+1:

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**